test/unit/io/sequence_file/sequence_file_integration_test.cpp

```diff
@@ -47,6 +47,40 @@ TEST(rows, assign_sequence_files)
     EXPECT_EQ(reinterpret_cast<std::ostringstream&>(fout.get_stream()).str(), output_comp);
 }
 
+TEST(rows, assign_sequence_files_read_blanks)
+{
+    std::string const input
+    {
+        ">TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN\n\n"
+        ">  Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    std::string const output_comp
+    {
+        ">TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        ">  Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
```

Comment:
wouldn't it be even more understandable if you use this as input AND output string. So one does not have to compare `input` and `output_comp` when looking at it but directly notices the "perfect roundtrip".

Or do you explicitly want to show that spaces in the sequence are still removed?

Reply:
Good point, will adjust. :+1: